src/main/java/org/folio/linked/data/service/resource/impl/ResourceEdgeServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package org.folio.linked.data.service.resource.impl;
+
+import java.util.Map;
+import java.util.function.Predicate;
+import org.folio.ld.dictionary.PredicateDictionary;
+import org.folio.ld.dictionary.ResourceTypeDictionary;
+import org.folio.linked.data.model.entity.Resource;
+import org.folio.linked.data.model.entity.ResourceEdge;
+import org.folio.linked.data.service.resource.ResourceEdgeService;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ResourceEdgeServiceImpl implements ResourceEdgeService {
+
+  private static final Predicate<ResourceEdge> IS_ADMIN_METADATA = edge -> edge.getPredicate().getUri()
+    .equals(PredicateDictionary.ADMIN_METADATA.getUri());
+
+  private final Map<ResourceTypeDictionary, Predicate<ResourceEdge>> edgesTobeCopied;
```

Comment:
We have two 'data sets' for one logical operation:
1) `IS_ADMIN_METADATA` and more predicates in future. Which actually will increase code lines number in the class each time we add one;
2) `edgesTobeCopied` map;

Can't we optimize it?
Just an option: what if we just keep a map of all Predicates which should be preserved per source resource type:
`Map<ResourceTypeDictionary, Set<PredicateDictionary>>`

When we just get a set of predicates for saving by given resource type, and use it as a filter for edges selection.

Benefits:
1) We store only one data set, which is extendable, but compact
2) `copyOutgoingEdges` method will be shorter
3) Shorter code in the class. Shorter is more readable

Reply:
I was thinking may be we have to handle more complex scenarios for copying edges. 
After reading your comment, I also agree that the complexity is not needed at least for now. When need arise in future, we will add more complex logic for copying the edges.